### ISSUE TITLE
[@typescript-eslint/no-shadow] Variables are incorrectly marked as shadowed

### ISSUE THREAD

**Issue by domoritz** [typescript-eslint/typescript-eslint#2469]
- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "no-shadow": "off",
  "@typescript-eslint/no-shadow": ["error"]
}
```

```TS
export function foo(bar: number, baz: (bar: number) => void) {}
```

**Expected Result**

no error

**Actual Result**

'bar' is already declared in the upper scope.eslintno-shadow

**Additional Info**

See https://github.com/eslint/eslint/issues/13649

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |

**Comment by kavsingh** [comment]
to add, same error as above occurs with config:
```json
{
  "no-shadow": "off",
  "@typescript-eslint/no-shadow": ["error", { "ignoreTypeValueShadow": true }]
}
```

an additional error case (with and without `ignoreTypeValueShadow` option) which used to work:
```ts
const _ = 'someVal';

const someFunc: {
  <T>(_: T): T;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
} = (t: any) => t;
```
errors:
![image](https://user-images.githubusercontent.com/813450/91916552-b8b40100-ecbd-11ea-84d2-7608cfdfd029.png)
